packages/code-du-travail-modeles/src/modeles/conventions/2596_coiffure/ineligibility.ts

```diff
@@ -0,0 +1,45 @@
+import { IneligibilityLegal } from "../../base";
+
+export class Ineligibility2596 extends IneligibilityLegal {
+  getIneligibility(
+    args: Record<string, string | undefined>
+  ): string | undefined {
+    if (args.typeContratTravail && args.typeContratTravail === "'cdd'") {
+      return "L’indemnité de licenciement n’est pas due pour les CDD et contrats de travail temporaires. Sous certaines conditions, le salarié peut avoir le droit à une indemnité de précarité.";
+    } else if (
+      args.licenciementFauteGrave &&
+      args.licenciementFauteGrave === "oui"
+    ) {
+      return "L’indemnité de licenciement n’est pas due en cas de faute grave (ou lourde). Lorsqu’il est invoqué, le motif de faute grave doit apparaître précisément dans le courrier. Reportez-vous à la lettre de notification de licenciement.";
```

Comment:
Dommage de devoir répéter ces règles. Est ce que l'on ne pourrait pas avoir des fonctions `getIneligilityCDD()`, `getIneligilityFauteGrave()`,... dans `IneligibilityLegal` qui serait appelé dans chaque cas spécifique de CC.

Reply:
oui j'hésitais à le séparer mais ca reviens un peu à refaire les steps côté modele. Ce que j'avais essayé de faire, c'était un système récursif pour éviter d'appeller l'inéligibité 2 fois, mais j'ai galéré pendant 2jours dessus sans succès.